Lighting/Src/lighting_controller.cpp

```diff
@@ -37,122 +38,69 @@ uint8_t bank_output_buffer[BANK_OUTPUT_BUFFER_SIZE];
-
-	// build strobe domain
-	rev3.set_domain_colour(CD_STROBE, WHITE, BRIGHTNESS_MAX);
-	rev3.add_led_to_cd(0, CD_STROBE);
-	rev3.add_led_to_cd(2, CD_STROBE);
-	rev3.add_led_to_cd(3, CD_STROBE);
-	rev3.add_led_to_cd(5, CD_STROBE);
-
+	// allow all of our domains
+	// comment any of these out to see the effect of allowing command domains
+	rev4.allow_domain(CD_MAIN);
+	rev4.allow_domain(CD_TAXI);
+	rev4.allow_domain(CD_BEACON);
+	rev4.allow_domain(CD_STROBE);
+	rev4.allow_domain(CD_LANDING);
+	rev4.allow_domain(CD_NAVIGATION);
+	rev4.allow_domain(CD_BRAKE);
+	rev4.allow_domain(CD_SEARCH);
+
+	rev4.activate_domain(CD_MAIN);
 	while (true) {
-		// Demo program to update LED colors & show control domain functionality
-
-		brightness_slow = brightness_slow + 1;
-		if (brightness_slow >= 20) {
-			brightness += brightness_direction;
-			brightness_slow = 0;
-		}
-		if (brightness >= 50) {
-			brightness = 50;
-			brightness_direction = -1;
-		} else if (brightness <= 0) {
-			brightness = 0;
-			brightness_direction = 1;
-		}
-
-		// Update red value
-		my_colour.red += red_direction;
-		if (my_colour.red >= 255) {
-			my_colour.red = 255;
-			red_direction = -1; // Start decreasing
-		} else if (my_colour.red <= 0) {
-			my_colour.red = 0;
-			red_direction = 1; // Start increasing
-		}
-
-		// Update green value
-		my_colour.green += green_direction;
-		if (my_colour.green >= 255) {
-			my_colour.green = 255;
-			green_direction = -1; // Start decreasing
-		} else if (my_colour.green <= 0) {
-			my_colour.green = 0;
-			green_direction = 1; // Start increasing
-		}
-
-		// Update blue value
-		my_colour.blue += blue_direction;
-		if (my_colour.blue >= 255) {
-			my_colour.blue = 255;
-			blue_direction = -1; // Start decreasing
-		} else if (my_colour.blue <= 0) {
-			my_colour.blue = 0;
-			blue_direction = 1; // Start increasing
-		}
-
-		// Add a small delay for smooth transitions
-		HAL_Delay(10); // Adjust this value for faster/slower fading
 
-		rev3.set_domain_colour(CD_MAIN, my_colour, brightness);
-		rev3.enable_control_domain(CD_MAIN);
 	}
 }
 
 LightingController::LightingController(uint8_t *dma_output_buffer,
-		uint8_t *bank_output_buffer, WS2812 *leds) {
+		uint8_t *bank_output_buffer, WS2812 *leds, TIM_HandleTypeDef *timer, uint16_t timer_channel) {
 	this->dma_buffer = dma_output_buffer;
 	this->bank_buffer = bank_output_buffer;
 	this->leds = leds;
+	this->htimx = timer;
+	this->tim_channel_x = timer_channel;
 	initialize_bank_buffer_on();
 	initialize_dma_buffer();
-
 	// Initialize all of the internal LED's as well
 	for (int i = 0; i < NUM_LEDS; ++i) {
-		this->leds[i].initialize_led_on(
+		this->leds[i].initialize_led_off(
 				bank_output_buffer + NUM_LEDS_PADDING * 24 + 24 * i);
 		this->domain_leds[CD_MAIN] |= 1 << i;	// this led index is now enabled
 	}
+
+	//enable beacon LEDs at the start to show that the drone is on
+	uint8_t beacon_leds = 1 + (1 << 2) + (1 << 3) + (1 << 5) + (1 << 6) + (1 << 8);
```

Comment:
Do you think it makse sense to eventually have an ENUM table for our LED's?

We don't need to add it this PR, but this would be good for a future PR.

```
typedef enum {
left_side_out = 1
ledt_side_in = 2
right_side_out = 3
right_side_in = 4
downwards_left_blank = x
} LedPosition_E
```
There has to be a better way to do this, i just don't know what it is right now.